server/src/main/java/org/cloudfoundry/identity/uaa/zone/MultitenantVmidentityClientDetailsService.java

```diff
@@ -333,4 +364,54 @@ private static ClientDetails mapOIDCClient(OIDCClient client, String tenant, Cas
         return list;
     }
 
+    private void sortClientDetails(List<ClientDetails> list, String sortBy, boolean ascending) {
```

Comment:
q: we have these compare function is at least 3 places? can we consolidate to some util ?
q: will these handle any null/optional ?